### ISSUE TITLE
helper function for validating a counterparty upgrade against the selfupgrade

### ISSUE THREAD

**Issue by charleenfei** [cosmos/ibc-go#3764]
In reference to this suggestion:

```go
    err := k.checkUpgradesForCompatibility(proposedUpgrade, counterpartyUpgrade)
    if err != nil {
        return types.NewUpgradeError(channel.UpgradeSequence, err)
    }
```

_Originally posted by @colin-axner in https://github.com/cosmos/ibc-go/pull/3671#discussion_r1211596403_

**Comment by colin-axner** [comment]
Also see comment [here](https://github.com/cosmos/ibc-go/pull/3628#discussion_r1213207844) for potential names of the three types of valdiation

**Comment by crodriguezvega** [comment]
Closed by #4016.
